---
title: "Lecture 20"
author: "Modified from R. Gray and L. Revell"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
  word_document: default
---

# Preamble

A few notes:

1. Don't forget: It's necesseary to run this before running any of the rest of the code, and you must also have a preamble coding chunk for your code to render correctly

2. A raw copy of this document is found on the class repository, titled `lecture_20.Rmd`

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)

# A new piece of code added to the preamble - packages are only installed if 
# they don't already exist in the R environment. 
# This is very useful for cutting down on rendering time

if(!require(ape)) install.packages("ape")
if(!require(phangorn)) install.packages("phangorn")
if(!require(phytools)) install.packages("phytools")
if(!require(geiger)) install.packages("geiger")
if(!require(phylogram)) install.packages("phylogram")
if(!require(remotes)) install.packages("remotes")
#remotes::install_github("GuangchuangYu/treeio")
if(!require(BiocManager)) install.packages("BiocManager")
if(!require(ggtree)) BiocManager::install("ggtree")
if(!require(DECIPHER)) BiocManager::install("DECIPHER")
if(!require(adegenet)) BiocManager::install("adegenet")
if(!require(viridis)) BiocManager::install("viridis")

library(ape)
library(phytools)
library(phangorn)
library(geiger)
library(phylogram)
library(seqinr)
library(adegenet)
library(ggtree)
#library(DECIPHER)
library(viridis)
library(ggplot2)

```

# Toy example of phylogenic tree

```{r, out.width="75%"}
# Read tree from string
text.string<-
    "(((((((cow, pig),whale),(bat,(lemur,human))),(robin,iguana)),coelacanth),gold_fish),shark);"
vert.tree<-read.tree(text=text.string)
plot(vert.tree,no.margin=TRUE,edge.width=2)
```

# Investigating Structure of Phylogenic Tree

```{r, out.width="75%"}
tree<-read.tree(text="(((A,B),(C,D)),E);")
plotTree(tree,offset=1)
tiplabels()
nodelabels()
```

# Comparison of Cytochrome b sequences from *Apodemus sylvaticus*
```{r, out.width="75%"}
data("woodmouse")
## generate distance matrices for each section of the alignment
dist1 <- dist.dna(woodmouse[, 1:482])
## build neighbor-joining trees
phy1 <- nj(dist1)
## plot as dendregram
plotTree(phy1,offset=1)
```

# Phylogenic Analysis - Fungal species

## Steps 1 & 2: Upload and align sequence data
```{r, include=FALSE, out.width="75%"}
# From Sequence analysis.R - R. Gray

# load the sequences from the file
# change "DNA" to "RNA" or "AA" if necessary
seqs <- readDNAStringSet("Amanita.txt", format = "fasta")

# nucleotide sequences need to be in the same orientation
# if they are not, then they can be reoriented (optional)
seqs <- OrientNucleotides(seqs)

# perform the alignment
aligned <- AlignSeqs(seqs)

# view the alignment in a browser (optional)
BrowseSeqs(aligned, highlight=0)

# write the alignment to a new FASTA file
writeXStringSet(aligned,
                file="Amanita_aligned.fasta")

# read in the aligned data
dna <- read.alignment("Amanita_aligned.fasta", format = "fasta")
```

## Step 3 - Calculate Distance Matrix

```{r, out.width="75%"}
# From Sequence analysis.R - R. Gray
# create a distance matrix for the alignment 
D <- dist.alignment(dna, matrix = "similarity")


temp <- as.data.frame(as.matrix(D))
table.paint(temp, cleg=0, clabel.row=.5, clabel.col=.5)+
  scale_color_viridis()
#darker shades of gray mean a larger distance 
# you can also make cool color plots but they're much more complicated because they use the image() function

# we can start to see a pattern because the data is ordered by year, 
# but we can't really make any conclusions yet

```

## Step 4 - Plotting Phylogenic Tree

```{r, out.width="75%"}
# From Sequence analysis.R - R. Gray
tre <- nj(D)
class(tre) #all trees created using {ape} package will be of class phylo

tre <- ladderize(tre)

# ~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~Base R plots ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

plot(tre, cex = 0.6)
title("similarity in Amanita (ITS)")


# or 
h_cluster <- hclust(D, method = "average", members = NULL) 
# method = average is used for UPGMA, members can be equal to NULL or a vector 
# with a length of size D
plot(h_cluster, cex = 0.6)
```


## Step 4 alternative - using 'ggtree' package to plot phlogeny
```{r, out.width="75%"}
# From Sequence analysis.R - R. Gray
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~ Tree Plotting in ggtree ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

# you can fan it out
ggtree(tre, yscale = "NA")+
  geom_tiplab(hjust = -0.3, size=4, align = TRUE)+
  xlim(0,0.5)

# plot a basic tree
ggtree(tre) +
  geom_tiplab(hjust = -0.3, size=4, align = TRUE)+
  xlim(0,0.5)


# ~~~~~~~~~~~~~~~~~~~~~~~~~~ Customize your trees ~~~~~~~~~~~~~~~~~~~~~~~~
# plot using ggtree and highlight clusters
# change the node values for your own data
ggtree(tre) + 
  geom_tiplab(hjust = -0.3, size=4, align = TRUE) + 
  geom_hilight(node=19, fill="purple", alpha = 0.2) + 
  geom_hilight(node=17, fill="dark green", alpha = 0.2) +
  geom_hilight(node=20, fill="gold", alpha = 0.2) +
  xlim(0,0.5) 

# highlight clusters and add a vertical line to group clusters
# change the node values for your own data
ggtree(tre) + 
  geom_tiplab(hjust = -0.3, size=4, align = TRUE) + 
  geom_hilight(node=19, fill="purple", alpha = 0.2) + 
  geom_hilight(node=17, fill="dark green", alpha = 0.2) +
  geom_hilight(node=20, fill="gold", alpha = 0.2) +
  geom_cladelabel(node=19, label=" Cluster 1", 
                  color="purple", offset=.1, barsize = 2,
                  fontsize = 5, align=TRUE, alpha = 0.5) + 
  geom_cladelabel(node=17, label=" Cluster 2", 
                  color="dark green", offset=.1, barsize = 2,
                  fontsize = 5, align=TRUE, alpha = 0.5) + 
  geom_cladelabel(node=20, label=" Cluster 3", 
                  color="gold", offset=.1, barsize = 2,
                  fontsize = 5, align=TRUE, alpha = 0.5) + 
  xlim(0,0.5) 
```  


## Optional - Analyse Alignment Data

Here you can plot your alignment data *with* your phylogenic tree - nifty!
```{r, out.width="75%"}
# From Sequence analysis.R - R. Gray
# ~~~~~~~~~~~~~~~~~~~~~~~~~~ Plot the alignment with the tree ~~~~~~~~~~~~~~~~~

# lets plot the alignment with the tree, to do this we first have to
# match the names to the tip labels
# set our tree into a new name
tre.new <- tre
# change tip labels to full alignment names
tre.new$tip.label <- aligned@ranges@NAMES

# plot the alignment 
msaplot(p=ggtree(tre.new), fasta="Amanita_aligned.fasta", 
        window=c(150, 175))+
  scale_fill_viridis_d(alpha = 0.8)
```

